cmd/kubeadm/app/cmd/upgrade/common.go

```diff
@@ -62,7 +62,8 @@ func isKubeadmConfigPresent(docmap kubeadmapi.DocumentMap) bool {
 // are loaded. This function allows the component configs to be loaded from a file that contains only them. If the file contains any kubeadm types
 // in it (API group "kubeadm.kubernetes.io" present), then the supplied file is treaded as a legacy reconfiguration style "--config" use and the
 // returned bool value is set to true (the only case to be done so).
-func loadConfig(cfgPath string, client clientset.Interface, skipComponentConfigs bool, printer output.Printer) (*kubeadmapi.InitConfiguration, bool, error) {
+// this function is defined as a variable so that it could be overwritten easily for test.
+var loadConfig = func(cfgPath string, client clientset.Interface, skipComponentConfigs bool, printer output.Printer) (*kubeadmapi.InitConfiguration, bool, error) {
 	// Used for info logs here
```

Comment:
a preferred way to do this is to allow the parent function to accept a function.
during trst it can be a test function and during real use it can accept the real loadConfig

Reply:
isn't it called in enforceRequirements?